### DISCUSSION TITLE
Is neuronal state (ie voltages) preserved between and within batches during SLAYER training?

### DISCUSSION THREAD

**Discussion by Michaeljurado42** [lava-nc/lava-dl#96] [General]
I am pretty sure that this would not be preserved, but I am asking just to make sure. What happens to the internal voltages within a batch and after a batch is used during Assistant training? Do the internal voltages just get reset to zero?

**Comment by bamsumit** [comment]
By default, they do not. But all slayer neurons have a parameter called `persistent_state`. It is by default `False`, but when `persistent_state=True`, the neurons preserve their previous states (voltages and currents for CUBA LIF). 
_NOTE: You have to ensure that the batch size is always the same between iterations, otherwise the previous state is ambiguous and gets reset._
